pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -80,6 +80,11 @@ const MaxReadThenCreateRetries = 30
 // need hardcoded defaults.
 const DefaultVolumeType = "gp2"
 
+// Amazon recommends having no more that 40 volumes attached to an instance,
+// and at least one of those is for the system root volume.
+// See http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html#linux-specific-volume-limits
+const DefaultMaxEBSVolumes = 39
```

Comment:
Yes, I think I do like the idea of having the scheduler restrict this.  I don't mind if you want to keep this here and raise a warning if we exceed 39 volumes... I expect that will save us a lot of time debugging :-)